File: solutions/d1000000/d1000000-solution.rs

use std::io;
use std::collections::BTreeMap;

#[allow(unused_macros)]
macro_rules! read {
    ($out:ident as $type:ty) => {
        let mut inner = String::new();
        std::io::stdin().read_line(&mut inner).expect("A String");
        let mut $out = inner.trim().parse::<$type>().expect("Parsable");
    };
}

#[allow(unused_macros)]
macro_rules! read_vec {
  ($out:ident as $type:ty) => {
    let mut inner = String::new();
    io::stdin().read_line(&mut inner).unwrap();
    #[warn(unused_variables)]
    #[warn(unused_variables)]
    let mut $out = inner
      .trim()
      .split_whitespace()
      .map(|s| s.parse::<$type>().unwrap())
      .collect::<Vec<$type>>();
  };
}

fn main(){
  let mut cas: i32 = 1;
  let mut solution = BTreeMap::new();
  read!(t as i32);
  while t > 0 {
    read!(n as i32);
    read_vec!(s as i32);    
    s.sort();
    
    let mut curr:i32 = 1;
    for dice in s {
      let dice: i32 = dice;
      if curr > dice {
        continue;
      }
      curr+=1;
    }
    solution.insert(cas,curr-1);
    cas += 1;
    t -= 1;
  }

  for (key, val) in solution.iter() {
    println!("Case #{:?}: {:?}", key, val);
  }
}

File: solutions/d1000000/d1000000-solution.cpp
#include <iostream>
#include <algorithm>
#include <vector>

using namespace std;

int main () {
   int t,n,cas=1;
   cin >> t;
   while (t--) {
       cin >> n;
       vector<int> s(n);
       for (int i = 0; i < n; i++) {
           cin >> s[i];
       }
       sort(s.begin(), s.end());
       int curr = 1;
       for (auto dice: s) {
           if (curr > dice) {
               continue;
           }
           curr++;
       }
       cout << "Case #" << cas++ << ": " << curr - 1 << endl;
   }
   return 0;
}
